### PR TITLE
feat(unlock): Add network description to network package and use those in checkout

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by searchableguy** [comment]
Let me know if any description needs to change.

File: packages/networks/src/networks/polygon.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export const polygon: NetworkConfig = {
       startBlock: 15714206,
     },
   ],
+  description: 'Layer 2 network. Cheaper transaction cost.',
```
Comment (julien51) [inline packages/networks/src/networks/polygon.ts:40]:
```suggestion
  description: 'Popular side chain network. Cheaper transaction cost.',
```

File: packages/networks/src/networks/xdai.ts [hunk 2]
```diff
@@ -37,5 +37,6 @@ export const xdai: NetworkConfig = {
       startBlock: 14521200,
     },
   ],
+  description: 'EVM compaitible network. Cheaper transaction cost.',
```
Comment (julien51) [inline packages/networks/src/networks/xdai.ts:40]:
```suggestion
  description: 'EVM compatible network whose base currency is a stable coin. Cheaper transaction cost.',
```

**Review by julien51 — APPROVED** [review]
Small changes, looking good overall!

File: packages/networks/src/networks/optimisticKovan.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ export const optimisticKovan: NetworkConfig = {
     symbol: 'Eth',
     decimals: 18,
   },
+  description: 'Layer 2 testing network. Do not use for production.',
```
Comment (searchableguy) [inline packages/networks/src/networks/optimisticKovan.ts:29]:
```suggestion
  description: 'Optimism testing network. Do not use for production.',
```

File: packages/networks/src/networks/mumbai.ts [hunk 4]
```diff
@@ -29,6 +29,7 @@ export const mumbai: NetworkConfig = {
   },
   startBlock: 26584912,
   previousDeploys: [],
+  description: 'Polygon testing network. Do not use for production',
```
Comment (searchableguy) [inline packages/networks/src/networks/mumbai.ts:32]:
```suggestion
  description: 'Polygon test network. Do not use for production',
```

**Review by julien51 — APPROVED** [review]

**Pull request merged**